### PR TITLE
Update Readme's macOS Instructions

### PR DESCRIPTION
**Pull request by LRFLEW**
Since I'm not the best at spelling, I'd like it if someone proof-read this. This replaces the old, very outdated instructions that were in the Readme previously. This includes both information about the Xcode project (which is the recommended way to build it) and CMake build instructions.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -85,22 +92,30 @@ msbuild openrct2.proj /t:g2
...
-```
-# Install libraries
-./install.sh
+### macOS:
+
+#### Xcode:
 
-# Build OpenRCT2
-./build.sh
+The reccomended way of building OpenRCT2 for macOS is with Xcode. Open the project file OpenRCT2.xcodeproj in Xcode and build from there. Building this way will handle the dependencies for you automatically. You can also invoke an Xcode build from the command line using `xcodebuild`.
```
Comment (janisozaur) [inline readme.md]: rec~~c~~omended

Comment (marijnvdwerf) [inline readme.md]: recommended

File: readme.md [hunk 2]
```diff
@@ -85,22 +92,30 @@ msbuild openrct2.proj /t:g2
...
+#### CMake:
+
+A command line version of OpenRCT2 can be built using CMake. This type of build requires you to provide the dependencies yourself. The supported method of doing this is with [Homebrew](http://brew.sh). Once you have Homebrew installed, you can download all the required libraries with this command:
+```
+brew install cmake openssl jansson libpng sdl2 sdl2_ttf speex
+```
+
+Once you have the dependencies installed, you can build the project using CMake using the following commands:
+```
+mkdir build && cd build
```
Comment (marijnvdwerf) [inline readme.md]: The `&&` is kinda annoying if you're using `fish` as shell. Why not put it on a separate line?

File: readme.md [hunk 3]
```diff
@@ -85,22 +92,30 @@ msbuild openrct2.proj /t:g2
...
 
-# Build OpenRCT2
-./build.sh
+The reccomended way of building OpenRCT2 for macOS is with Xcode. Open the project file OpenRCT2.xcodeproj in Xcode and build from there. Building this way will handle the dependencies for you automatically. You can also invoke an Xcode build from the command line using `xcodebuild`.
 
-# Run the game
-./openrct2
+#### CMake:
+
+A command line version of OpenRCT2 can be built using CMake. This type of build requires you to provide the dependencies yourself. The supported method of doing this is with [Homebrew](http://brew.sh). Once you have Homebrew installed, you can download all the required libraries with this command:
```
Comment (janisozaur) [inline readme.md]: What do you mean by "command line version"? Does Xcode strip away `stdin`, `stdout` and `stderr` and ability to pass arguments?

File: readme.md [hunk 4]
```diff
@@ -56,10 +56,17 @@ OpenRCT2 requires original files of RollerCoaster Tycoon 2 to play. It can be bo
 - [7-Zip](http://www.7-zip.org/) (for deployment only)
 - [NSIS](http://nsis.sourceforge.net/) (for deployment only)
 
-### Mac:
+### macOS:
+
+- Xcode 8
+
+The program can also be built as a command line program using CMake. This type of build requires:
```
Comment (LRFLEW) [inline readme.md]: The primary difference between the Xcode build and the CMake build is that the Xcode build creates a self-contained app bundle, whereas the CMake build just produces the binary. In general, the binaries are only meant to be run from the command line. Do you have any suggestions on how I can word this better?

Comment (marijnvdwerf) [inline readme.md]: A normal app on macOS is an ".app" bundle. You normally don't run a binary directly, except when doing CLI stuff.

Comment (janisozaur) [inline readme.md]: then say Xcode creates app bundle

File: readme.md (hunk 2 above)
Comment (LRFLEW) [inline readme.md]: There isn't really a big reason; I copied this from the last time I used CMake. I personally use `&&` for this because if (for some reason) the mkdir command fails, I don't want to try to cd into a directory that may not be there. I will change this back.

**Comment by janisozaur** [comment]
I'd like to review it tomorrow, don't merge yet

File: readme.md (hunk 4 above)
Comment (LRFLEW) [inline readme.md]: I went ahead and added a message to the Xcode description mentioning the app bundle. I still think having the explanation that the CMake build create a "command line" binary is useful. Is there anything else I should add/change to make it clearer, though?

Comment (IntelOrca) [inline readme.md]: console application?

File: readme.md [hunk 5]
```diff
@@ -85,22 +92,31 @@ msbuild openrct2.proj /t:g2
...
+cd build
+cmake ..
+make
+ln -s ../data data
 ```
+Then you can run the game by running `./openrct2`.
 
 ### Linux:
-We support native builds for Linux. x86 builds, by default, make use of pieces of original code, other builds are completely standalone.
+We support native builds for Linux.
```
Comment (IntelOrca) [inline readme.md]: > We support native builds for Linux.

You can probably scrap this altogether.

File: readme.md (hunk 4 above)
Comment (marijnvdwerf) [inline readme.md]: This is the 'Console' application.
image

Xcode calls it a 'Command Line Tool', so that term will probably the standard way to call it.
image

Comment (LRFLEW) [inline readme.md]: That is a good example of how "Command Line" is used in the macOS space. The only issue with using that verbatim is that OpenRCT2 can't really be described as a "Tool". This is why I kept the descriptor "Command Line", but changed the last part. I could try to reword it to use "Command Line" as a noun (ex. "OpenRCT2 can be built as a binary that can be run from the command line..."), but it could make the phrasing a bit awkward. I can also change "Program" to "Application" or something else.

File: readme.md [hunk 6]
```diff
@@ -85,23 +91,28 @@ msbuild openrct2.proj /t:g2
...
-# Install libraries
-./install.sh
+### macOS:
+#### Xcode:
+The recommended way of building OpenRCT2 for macOS is with Xcode. The Xcode build will create a self-contained application bundles which include all the necessary game files and dependencies. Open the project file OpenRCT2.xcodeproj in Xcode and build from there. Building this way will handle the dependencies for you automatically. You can also invoke an Xcode build from the command line using `xcodebuild`.
 
-# Build OpenRCT2
-./build.sh
+#### CMake:
+A command line version of OpenRCT2 can be built using CMake. This type of build requires you to provide the dependencies yourself. The supported method of doing this is with [Homebrew](http://brew.sh). Once you have Homebrew installed, you can download all the required libraries with this command:
```
Comment (janisozaur) [inline readme.md]: I still don't think it is valid description. From a perspective of a user reading this readme, I would assume this is a different kind of build than Xcode one. As in: Xcode is not able of performing any command line interactions, which simply is not true.

It would be better if you simply pointed out that this does not build "self-contained application bundles", but do check it is the case with `make package`.

Comment (LRFLEW) [inline readme.md]: To be fair, I accidentally pushed this change, and I was going to review/discuss it further 😜 

While it is true that the Xcode build *can* be run from the command line, it's really not easy or intuitive. Plus, the CMake build is much more "command line" due to how it looks for the data files compared to when it's in the app bundle. I still think "command line" is the best way to describe it without adding a paragraph spelling out what an app bundle is and all the workarounds added to get the Xcode build to work without requiring the user to install dependencies system-wide.

Also, `make package` doesn't make an app bundle. I've tried it previously.

File: readme.md [hunk 7]
```diff
@@ -56,10 +56,16 @@ OpenRCT2 requires original files of RollerCoaster Tycoon 2 to play. It can be bo
 - [7-Zip](http://www.7-zip.org/) (for deployment only)
 - [NSIS](http://nsis.sourceforge.net/) (for deployment only)
 
-### Mac:
+### macOS:
+- Xcode 8
+
+The program can also be built as a command line program using CMake. This type of build requires:
```
Comment (LRFLEW) [inline readme.md]: It's worth flagging this so I don't forget, but the same "command line" phrasing was used here (so if we change it, we should change it here too)

**Comment by marijnvdwerf** [comment]
For what it's worth, I agree with LRFLEW. The text makes sense in a macOS
context.
Op vr 21 okt. 2016 om 22:45 schreef LRFLEW <notifications@github.com>
